.changeset/ninety-onions-bow.md

```diff
@@ -0,0 +1,5 @@
+---
+'create-astro': minor
+---
+
+Automatically installs the required dependencies to run the `astro check` command when TypeScript mode is selected.
```

Comment:
So to clarify, `astro-check` is added even when someone says "No I don't intend to write TypeScript?" Which would be every single new project?

In that case, the above is fine, but it's really just:

```suggestion
Automatically installs the required dependencies to run the `astro check` command.
```
Am I missing something? I'm not sure it matters that it happens when you choose which TypeScript mode, since the choice you make doesn't affect whether or not it's installed? If so, I think just THAT it installs the dependencies seems adequate. But, either would be fine!

Reply:
No, if user don't intend to write TypeScript `astro check` will not be added.

1. If user does not choose to use TypeScript, nothing happens **(no behavior change)**
2. If user choose TypeScript, `astro check` will be added

Reply:
> I do agree. But I have to adjust some wording at the end because astro check will be applied to all TypeScript options (relaxed, strict, strictest).

This was your reply to my other comment, where I suggested mentioning only `strict` and `strictest`.

Reply:
The intent is that the `astro check` code is added only if the user answers "Yes" to "Do you plan to write TypeScript?". Even if they choose "relaxed" they will still get the `astro check` code.

![Do you plan to write TypeScript? Yes or No](https://github.com/withastro/astro/assets/7118177/e54ccf43-9c69-451e-bed1-a481bb05896d)

Reply:
This is arguably a lot of discussion for something not super important. :sweat_smile: 

Just checking, we have the templates `base`, `strict` and `strictest`?

And my original suggestion was because selecting that the user does intend to write TypeScript is different from "when configuring TypeScript" because you're ALWAYS configuring TS at this step. (You're just not always choosing one of the stricter settings.) Similarly, it's not "when choosing a TS mode" because you're always choosing a TS mode at this point, just not always one that installs `astro check`.

This is why I stand by this option as being the most informative:

> Automatically installs the required dependencies to run the `astro check` command when `strict` or `strictest` TypeScript mode is selected.

However, if it's hidden from the user that their choice selects one of those two modes by name (e.g. by using different words), then a more user-facing alternative could be something like:

> Automatically installs the required dependencies to run the `astro check` command when the user indicates they plan to write TypeScript.

Reply:
I had hoped my review would reply to this comment, but it didn't. I like your final suggestion here. Resolved in https://github.com/withastro/astro/pull/8853#discussion_r1369276927